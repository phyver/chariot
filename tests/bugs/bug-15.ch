data nat where
    | Zero : nat

data c('a) where
    | C : 'a -> c('a)

codata nuY('x) where
    | Nu : nuY('x) -> c('x)

codata stream4 where
    | D4 : stream4 -> nuY(stream4)


val zeros4 = { D4 = { Nu = C zeros4 } }

val convert4 { D4 = { Nu = C s } } = 0

:set verbose 10
:show functions
:set verbose 1

-- ERROR
val test12 = convert4 zeros4
:unfold test12 , 0


