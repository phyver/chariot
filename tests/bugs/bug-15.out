>> option verbose is set to 1
>> inductive type nat was succesfully defined
>> inductive type c was succesfully defined
>> coinductive type nuY was succesfully defined
>> coinductive type stream4 was succesfully defined
>> Typing for zeros4 successful
>> the definition for zeros4 is complete
>> the definition zeros4 is provably correct

>> Typing for convert4 successful
>> the definition for convert4 is complete
>> the definition convert4 is provably correct

>> option verbose is set to 10
>> functions in environment:
val
   zeros4 : stream4
    | zeros4 = { D4 = { Nu = C¹ zeros4 }⁰ }²
>> 
>> corresponding case and struct form:
>> { D4 = { Nu = C zeros4 } }

val
   convert4 : stream4 → nat
    | convert4 { D4 = { Nu = C¹ s }⁰ }² = 0
>> 
>> corresponding case and struct form:
>> x₁ -> 
>>   case x₁.D4.Nu of
>>     C x₂  ->  0

>> option verbose is set to 1
>> Typing for test12 successful
>> the definition for test12 is complete
>> the definition test12 is provably correct

>> term: test12
>> of type: nat
>> result: 0

